src/main/java/com/keniu/services/OrderServiceImpl.java

```diff
@@ -0,0 +1,118 @@
+import com.keniu.mappers.OrderMapper;
+import com.keniu.models.CartItem;
+import com.keniu.models.Order;
+import com.keniu.models.OrderItem;
+import com.keniu.models.ShoppingCart;
+import com.keniu.models.Status;
+import com.keniu.models.User;
+import com.keniu.repositories.OrderRepository;
+import com.keniu.repositories.ShoppingCartRepository;
+import jakarta.transaction.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+@Transactional
+public class OrderServiceImpl implements OrderService {
+
+    private final OrderRepository orderRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+
+    @Override
+    @Transactional
+    public OrderDto save(User user, CreateOrderRequestDto createOrderRequestDto) {
+        Order order = createOrderFromDto(createOrderRequestDto, user);
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    @Override
+    public Page<OrderDto> findAll(User user, Pageable pageable) {
+        return orderRepository.findByUserId(user.getId(), pageable).map(orderMapper::toDto);
+    }
+
+    @Override
+    public OrderDto update(Long id, UpdateOrderRequestDto updateOrderRequestDto) {
+        Order order = findOrderById(id);
+        orderMapper.updateOrderFromDto(updateOrderRequestDto, order);
+        return orderMapper.toDto(orderRepository.save(order));
+    }
+
+    public Set<OrderItemDto> findItemsForOrder(Long id) {
+        Order order = findOrderById(id);
+        return order.getOrderItems().stream()
+            .map(orderItemMapper::toDto)
+            .collect(Collectors.toSet());
+    }
+
+    @Override
+    public OrderItemDto findItemForOrder(Long orderId, Long itemId) {
+        Order order = findOrderById(orderId);
+        return order.getOrderItems().stream()
+            .filter(item -> item.getId().equals(itemId))
+            .map(orderItemMapper::toDto)
+            .findFirst()
+            .orElseThrow(() -> new EntityNotFoundException("Can't find item by id " + itemId));
+    }
+
+    private Order createOrderFromDto(CreateOrderRequestDto createOrderRequestDto, User user) {
+        Order order = orderMapper.toModel(createOrderRequestDto);
+        order.setUser(user);
+        order.setStatus(Status.PENDING);
+        ShoppingCart shoppingCart = findShoppingCartByUser(user);
+        Set<OrderItem> orderItems = createOrderItems(shoppingCart);
+        BigDecimal total = calculateTotal(orderItems);
+
+        order.setOrderItems(orderItems);
+        order.setTotal(total);
+        order.setOrderDate(LocalDateTime.now());
```

Comment:
JFYI you can use the CreationTimestamp annotation

Reply:
Good idea!